cmd/nerdctl/exec_test.go

```diff
@@ -44,3 +44,16 @@ func TestExecWithDoubleDash(t *testing.T) {
 
 	base.Cmd("exec", testContainer, "--", "echo", "success").AssertOutExactly("success\n")
 }
+
+func TestExecWithUser(t *testing.T) {
+	t.Parallel()
+	base := testutil.NewBase(t)
+	const testContainer = "nerdctl-test-exec"
+	defer base.Cmd("rm", "-f", testContainer).Run()
+
+	base.Cmd("run", "-d", "--name", testContainer, testutil.CommonImage, "sleep", "1h").AssertOK()
+
+	base.Cmd("exec", testContainer, "id").AssertOutContains("uid=0(root) gid=0(root)")
+	base.Cmd("exec", "--user", "1000", testContainer, "id").AssertOutContains("uid=1000 gid=0")
+	base.Cmd("exec", "--user", "1000:1000", testContainer, "id").AssertOutContains("uid=1000 gid=1000")
+}
```

Comment:
We need an equivalent of ` TestRunUserGID` too
 https://github.com/containerd/nerdctl/blob/c3a5ce90aac4fdb22c455b54671d1f897dc2b9e0/cmd/nerdctl/run_user_linux_test.go#L25-L47